packages/deprecation-crawler/src/lib/deprecation-crawler.ts

```diff
@@ -1,19 +1,24 @@
 import { getConfig } from './config';
 import { CrawledRelease } from './models';
 import { stripIndent } from 'common-tags';
-import { branchHasChanges, run } from './utils';
+import {
+  branchHasChanges,
+  isCrawlerModeCi,
+  isCrawlerModeSandbox,
+  run,
+} from './utils';
 import { logError } from './log';
 import { checkout } from './tasks/checkout';
 import { crawl } from './tasks/crawl';
 import { updateRepository } from './tasks/update-repository';
 import { addGroups } from './tasks/add-groups';
 import { generateOutput } from './tasks/generate-output';
 import { commitChanges } from './tasks/commit-changes';
-import { CRAWLER_MODES } from './constants';
 
 (async () => {
-  await guardAgainstDirtyRepo();
-
+  if (isCrawlerModeCi()) {
```

Comment:
We now have 2 checks for `guardAgainstDirtyRepo`.
This new one, and the existing `if (isCrawlerModeSandbox()) {` inside the `guardAgainstDirtyRepo` method.
Can you remove one of the conditions please?

Reply:
This is very interesting as it forces me to think about a standardized way of where to put "feature toggles"...
ATM I prefer to have the behavior e.g. `guardAgainstDirtyRepo` isolated in a function and the usage separate.
This obviously leads to repetitive checks when using it. :D

Opinions?

Reply:
👍 we can do that.
If we got a better idea later, we can always refactor 😅